xonsh/events.py

```diff
@@ -0,0 +1,101 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.doc("Comes with eggs")
+"""
+
+
+class Event(set):
```

Comment:
better to inherit from `collections.abc.Set` 
although i'm pro ordered collection aswell